iocage_lib/ioc_start.py

```diff
@@ -525,30 +520,66 @@ def __start_jail__(self):
         self.start_generate_resolv()
         self.start_copy_localtime()
         # This needs to be a list.
-        exec_start = self.conf["exec_start"].split()
+        exec_start = self.conf['exec_start'].split()
+
+        with open(
+            os.path.join(self.iocroot, 'log', f'{self.uuid}-console.log'), 'a'
+        ) as f:
+            success, output = iocage_lib.ioc_common.safe_checkoutput(
+                ['setfib', self.exec_fib, 'jexec', f'ioc-{self.uuid}']
+                + exec_start, stderr=su.STDOUT
+            )
 
-        with open("{}/log/{}-console.log".format(self.iocroot,
-                                                 self.uuid), "a") as f:
-            services = su.check_call(["setfib", self.exec_fib, "jexec",
-                                      f"ioc-{self.uuid}"] + exec_start,
-                                     stdout=f, stderr=su.PIPE)
+            f.write(output)
 
-        if services:
-            msg = "  + Starting services FAILED"
+        if not success:
+            msg = f'  + Starting services FAILED\nERROR:\n{output}\n\n' \
+                'exec_start failed to execute.\nJail failed to start'
             iocage_lib.ioc_common.logit({
-                "level": "ERROR",
-                "message": msg
+                'level': 'EXCEPTION',
+                'message': msg
             },
                 _callback=self.callback,
-                silent=self.silent)
+                silent=self.silent
+            )
+
+            iocage_lib.ioc_stop.IOCStop(
```

Comment:
This needs to run before the logit, as otherwise the program will be dead.

Reply:
Good catch ;)